edbterraform/data/terraform/aws/modules/machine/main.tf

```
@@ -31,6 +31,21 @@ resource "aws_instance" "machine" {
   subnet_id              = var.subnet_id
   vpc_security_group_ids = flatten([var.custom_security_group_ids, module.machine_ports.security_group_ids])
 
+  dynamic "instance_market_options" {
+    for_each = var.machine.spec.spot_max_price[*]
+    content {
+      market_type = "spot"
+      dynamic "spot_options" {
+        for_each = var.machine.spec.spot_max_price[*]
+        content {
+          instance_interruption_behavior = "terminate"
```

During provision I faced another error:
```
╷
│ Error: creating EC2 Instance: InvalidParameterCombination: The terminate InstanceInterruptionBehavior is not supported when requestType is set to persistent.
│       status code: 400, request id: 103e214c-3ece-43ea-abe3-9fcc5ac65948
│ 
│   with module.machine_us_east_1["pg1"].aws_instance.machine,
│   on modules/machine/main.tf line 27, in resource "aws_instance" "machine":
│   27: resource "aws_instance" "machine" {
│ 
╵
```
I set `instance_interruption_behavior = "hibernate"` instead of `terminate`.
There is also `stop`, the difference being that if the instance stops, some of ram is persisted to storage if possible. If an instance does not support `hibernate` then it defaults to stopping the instance.

More details below:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/hibernate-spot-instances.html
```suggestion
          instance_interruption_behavior = "hibernate"
```

---

Ok, changed default to hibernate.  Push forced.